analysis/emissionscost.py

```diff
@@ -208,49 +218,70 @@ def shadowcost_tariff(
                     emissions_path=None,
                     flex_capacity=continuous_flexibility,
                     rte=1.0,
-                    min_onsteps=max(int(len(aef) * (system_uptime)), 1),  # assuming daily cycles
+                    min_onsteps=max(int(len(emis) * (system_uptime)), 1),  # assuming daily cycles
                     uptime_equality=uptime_equality,
 )
     flex_cost.build()
     flex_cost.solve(threads = threads)
     # Calculate the net facility load
     net_facility_load_costopt = idxparam_value(flex_cost.model.net_facility_load)
 
-    # calculate the total cost and total emissions using the output power
-    # cost_optimal_cost = tariff@net_facility_load_costopt
-    cost_optimal_cost = flex_cost.model.total_flex_cost_signal()
+    # (1) get the charge dictionary
+    charge_dict = costs.get_charge_dict(
+        startdate_dt, enddate_dt, tariff_data, resolution="1h"
+    )
+
+    # (2) set up consumption data dictionary
+    consumption_data_dict_costopt = {"electric": net_facility_load_costopt}
+
+    cost_optimal_cost, _ = costs.calculate_cost(
```

Comment:
This may be including `customer` charges (depending on if `keep_fixed_charges` is `True` or `False`), so it would be good to explicitly pass `keep_fixed_charges=False` in case the default value of that flag changes